hs/src/Reach/Debugger.hs

```diff
@@ -0,0 +1,100 @@
+{-# OPTIONS_GHC -Wno-missing-export-lists #-}
+
+module Reach.Debugger where
+
+import Control.Monad.Reader
+
+import qualified Data.Map.Strict as M
+import Reach.AST.Base
+import Reach.AST.DLBase
+import Reach.AST.LL
+
+-- NOTE: let several base types be Integer/String as a placeholder for now
+
+type ConsensusEnv = M.Map DLVar DLVal
+
+type Balances = [Double]
+
+type Ledger = M.Map Account Balances
+
+type Frontend = ( Account, SLPart, M.Map DLVar DLVal, [ DLTail ] )
+
+type Frontends = [ Frontend ]
+
+type NewPartActions = M.Map SLPart [ DLTail ]
+
+type State = (Ledger, ConsensusEnv, Frontends, NewPartActions, Action)
+
+-- possible backend actions
+data Action
+  = ConsAction LLConsensus
+  | StepAction LLStep
```

Comment:
I think that the action would be something like `compute` and it means "do one step of computation", however, maybe you could make it so that all such boring "compute" actions are ignored and not presented to the user... we only present to them the options that they can control? It might be tedious to press "next" 17 times.